parsl/executors/globus_compute.py

```diff
@@ -0,0 +1,162 @@
+
+        batch_size:
+            the maximum number of tasks to coalesce before
+            sending upstream [min: 1, default: 128]
+
+        amqp_port:
+            Port to use when connecting to results queue. Note that the
+            Compute web services only support 5671, 5672, and 443.
+
+        client:
+            instance of globus_compute_sdk.Client to be used by the executor.
+            If not provided, the executor will instantiate one with default arguments.
+
+        kwargs:
+            Other kwargs listed will be passed through to globus_compute_sdk.Executor
+            as is
+        """
+        super().__init__()
+        self.endpoint_id = endpoint_id
+        self.task_group_id = task_group_id
+        self.resource_specification = resource_specification
+        self.user_endpoint_config = user_endpoint_config
+        self.label = label
+        self.batch_size = batch_size
+        self.amqp_port = amqp_port
+        self.client = client
+
+        if not _globus_compute_enabled:
+            raise OptionalModuleMissing(
+                ['globus-compute-sdk'],
+                "GlobusComputeExecutor requires globus-compute-sdk installed"
+            )
+
+        self._executor: Executor = Executor(
+            endpoint_id=endpoint_id,
+            task_group_id=task_group_id,
+            resource_specification=resource_specification,
+            user_endpoint_config=user_endpoint_config,
+            label=label,
+            batch_size=batch_size,
+            amqp_port=amqp_port,
+            client=self.client,
+            **kwargs
+        )
+
+    def start(self) -> None:
+        """Empty function
+        """
+        pass
+
+    def submit(self, func: Callable, resource_specification: Dict[str, Any], *args: Any, **kwargs: Any) -> Future:
+        """ Submit fn to globus-compute
+
+
+        Parameters
+        ----------
+
+        func: Callable
+            Python function to execute remotely
+
+        resource_specification: Dict[str, Any]
+            Resource specification can be used specify MPI resources required by MPI applications on
+            Endpoints configured to use globus compute's MPIEngine. GCE also accepts *user_endpoint_config*
+            to configure endpoints when the endpoint is a `Multi-User Endpoint
+            <https://globus-compute.readthedocs.io/en/latest/endpoints/endpoints.html#templating-endpoint-configuration>`_
+
+        args:
+            Args to pass to the function
+
+        kwargs:
+            kwargs to pass to the function
+
+        Returns
+        -------
+
+        Future
+        """
+        self._executor.resource_specification = resource_specification or self.resource_specification
+        # Pop user_endpoint_config since it is illegal in resource_spec for globus_compute
+        self._executor.user_endpoint_config = resource_specification.pop('user_endpoint_config', self.user_endpoint_config)
```

Comment:
Should the `pop` happen before you assign `resource_specification` to `self._executor.resource_specification`?

Reply:
I think: the `pop` live-edits the value previously assigned to self._executor.resource_specification - because it is assignment of mutable objects, not of values.

But following that same reasoning, @yadudoc does this mean this code is editing the resource_specification dict provided by the user? (at some arbitrary point in the future after the user invokes an app...?)

Reply:
Oh, right!  Of course. I'm mixing up my languages.

Reply:
I've updated this section to pop `user_endpoint_config` from a copy of `resource_specification` to avoid modifying the user-supplied dict.